funnel/views/sms_events.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict
+import json
+
+from flask import request, url_for
+
+from twilio.request_validator import RequestValidator
+
+from baseframe import statsd
+from coaster.views import render_with
+
+from .. import app
+from ..models import SMS_STATUS, SMSMessage, db
+from ..transports.sms import TwilioSmsResponse
+
+
+class TwilioMessageValidator:
+    """ Validator for Twilio SMS Messages """
+
+    def __init__(self):
+        """ Just setup the Object variables, but it will be used later. """
+        self.url = None
+        self.validator = None
+
+    def check(self, signature: str, message: Dict) -> bool:
+        """
+        Check if the incoming message can be validated as coming from Twilio
+        :param signature Message Signature
+        :param message   Incoming Message Payload
+        """
+        if not self.url:
+            self.url = url_for('process_twilio_event', _external=True)
+            self.validator = RequestValidator(app.config['SMS_TWILIO_TOKEN'])
+
+        # Let us now validate and reject if signatures mismatch. See this:
+        # https://www.twilio.com/docs/usage/security#validating-requests
+        return self.validator.validate(self.url, message, signature)
+
+
+# Unexposed global validator
+twilio_validator = TwilioMessageValidator()
+
+
+@app.route('/api/1/sms/twilio_event', methods=['POST'])
+@render_with(template=None, json=True)
+def process_twilio_event():
+    """ Processes SMS callback events from Twilio """
+
+    # Register the fact that we got a Twilio SMS event.
+    # If there are too many rejects, then most likely a hack attempt.
+    statsd.incr('sms_message.twilio_event.received')
```

Comment:
Let's prefix these with `phone_number.` for the forthcoming `phone_number` table. So `phone_number.sms.twilio_event.received`